File: 268-missing-number/268-missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
       //best approach xor
            int n=nums.size();
          int v,index;
            for(v=0 ,index=0; index<n;index++){
                    v=(v^index) ^nums[index];
            }
            return v^index;
    }
};

File: 268-missing-number/268-missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
       int s=0;
            int n=nums.length;
                                 
            for(int i=0;i<n;i++){
                    s+=nums[i];
                    //System.out.println(s)
            }
            
            return (n*(n+1))/2 - s;
    }
}